dali/pipeline/operators/operator.h

```diff
@@ -197,32 +209,48 @@ class Operator<CPUBackend> : public OperatorBase {
   inline ~Operator() noexcept(false) override {}
 
   using OperatorBase::Run;
-  void Run(SampleWorkspace *ws) override {
+
+  void Run(HostWorkspace *ws) override {
     CheckInputLayouts(ws, spec_);
     SetupSharedSampleParams(ws);
     for (int i = 0; i < input_sets_; ++i) {
       RunImpl(ws, i);
     }
+
+    ws->GetThreadPool().WaitForWork();
   }
 
   /**
    * @brief Legacy implementation of CPU operator using per-sample approach
    *
-   * Usage of this API will be deprecated.
+   * Usage of this API is deprecated. For CPU Ops `void RunImpl(HostWorkspace *ws, int idx)`
+   * should be overrided instead.
    */
   virtual void RunImpl(SampleWorkspace *ws, int idx = 0) {}
 
   /**
    * @brief Implementation of the operator - to be implemented by derived ops.
    */
   virtual void RunImpl(HostWorkspace *ws, int idx = 0) {
-    DALI_ENFORCE(false, "Not implemented yet");
+    // This is implemented, as a default, using the RunImpl that accepts SampleWorkspace,
+    // allowing for fallback to old per-sample implementations.
+
+    for (int data_idx = 0; data_idx < batch_size_; ++data_idx) {
+      auto &tp = ws->GetThreadPool();
```

Comment:
`thread_pool` would be easier to read, I think

Reply:
Done